tiledb/api/c_api/array_schema/array_schema_api_internal.h

```diff
@@ -0,0 +1,321 @@
+/**
+ * @file tiledb/api/c_api/array_schema/array_schema_api_internal.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the internals of the array_schema section of the C API.
+ */
+
+#ifndef TILEDB_CAPI_ARRAY_SCHEMA_INTERNAL_H
+#define TILEDB_CAPI_ARRAY_SCHEMA_INTERNAL_H
+
+#include "array_schema_api_external.h"
+
+#include "tiledb/api/c_api_support/handle/handle.h"
+#include "tiledb/common/common.h"
+#include "tiledb/sm/array_schema/array_schema.h"
+#include "tiledb/sm/enums/array_type.h"
+
+using namespace tiledb::sm;
+
+/** Handle `struct` for API ArraySchema objects. */
+struct tiledb_array_schema_handle_t
+    : public tiledb::api::CAPIHandle<tiledb_array_schema_handle_t> {
+  /** Type name */
+  static constexpr std::string_view object_type_name{"array_schema"};
+
+ private:
+  using array_schema_type = shared_ptr<tiledb::sm::ArraySchema>;
+  array_schema_type array_schema_;
+
+  /**
+   * Size type for the number of attributes of an array and for attribute
+   * indices.
+   */
+  using attribute_size_type = unsigned int;
```

Comment:
What is the purpose of these type aliases?

Reply:
To clarify what exactly the size types refer to, rather than just using `unsigned int`. 
Ref: https://github.com/TileDB-Inc/TileDB/pull/3194

Reply:
I redefined them here for clarification's sake but in retrospect, I think `ArraySchema::alias` is just fine.